lib/base/utility.cpp

```diff
@@ -1939,3 +1941,31 @@ String Utility::GetIcingaDataPath(void)
 }
 
 #endif /* _WIN32 */
+
+bool Utility::IsAbsolutePath(const String& path)
+{
+#ifndef _WIN32
+	return (path.GetLength() > 0 && path[0] == '/');
+#else /* _WIN32 */
+	return !PathIsRelative(path.CStr());
+#endif /* _WIN32 */
+}
+
+String Utility::ConcatPath(std::initializer_list<std::string> components)
+{
+	boost::filesystem::path result;
+	for (const auto& component : components) {
+		result.append(component);
+	}
+	return result.string();
```

Comment:
Please re-implement this using existing code and helpers for creating a plugin path.